src/main/java/io/reactivex/internal/schedulers/SchedulerPoolFactory.java

```diff
@@ -81,61 +82,77 @@ public static void start() {
      * Stops the purge thread.
      */
     public static void shutdown() {
-        ScheduledExecutorService exec = PURGE_THREAD.get();
+        ScheduledExecutorService exec = PURGE_THREAD.getAndSet(null);
         if (exec != null) {
             exec.shutdownNow();
         }
         POOLS.clear();
     }
 
     static {
-        boolean purgeEnable = true;
-        int purgePeriod = 1;
-
         Properties properties = System.getProperties();
 
-        if (properties.containsKey(PURGE_ENABLED_KEY)) {
-            purgeEnable = Boolean.getBoolean(PURGE_ENABLED_KEY);
-        }
+        PurgeProperties pp = new PurgeProperties();
```

Comment:
Factored out to be able to test parameter loading. This static code block executes once per class init.